packages/react-query/src/shared/hooks/createHooksInternal.tsx

```diff
@@ -416,30 +421,35 @@ export function createRootHooks<
     // request option should take priority over global
     const shouldAbortOnUnmount = opts?.trpc?.abortOnUnmount ?? abortOnUnmount;
 
-    const hook = __useInfiniteQuery({
-      ...ssrOpts,
-      queryKey: queryKey as any,
-      queryFn: (queryFunctionContext) => {
-        const actualOpts = {
-          ...ssrOpts,
-          trpc: {
-            ...ssrOpts?.trpc,
-            ...(shouldAbortOnUnmount
-              ? { signal: queryFunctionContext.signal }
-              : {}),
-          },
-        };
-
-        return client.query(
-          ...getClientArgs(
-            queryKey,
-            actualOpts,
-            queryFunctionContext.pageParam ?? opts?.initialCursor,
-          ),
-        );
+    const hook = __useInfiniteQuery(
+      {
+        ...ssrOpts,
+        queryKey: queryKey as any,
+        defaultPageParam: opts.defaultPageParam,
+        getNextPageParam: opts.getNextPageParam,
+        getPreviousPageParam: opts.getPreviousPageParam,
+        queryFn: (queryFunctionContext) => {
+          const actualOpts = {
+            ...ssrOpts,
+            trpc: {
+              ...ssrOpts?.trpc,
+              ...(shouldAbortOnUnmount
+                ? { signal: queryFunctionContext.signal }
+                : {}),
+            },
+          };
+
+          return client.query(
+            ...getClientArgs(
+              queryKey,
+              actualOpts,
+              queryFunctionContext.pageParam ?? opts.defaultPageParam,
+            ),
+          );
+        },
       },
-      context: ReactQueryContext,
-    }) as UseTRPCInfiniteQueryResult<unknown, TError>;
+      queryClient,
```

Comment:
https://tanstack.com/query/v5/docs/react/guides/migrating-to-v5#removed-custom-context-prop-in-favor-of-custom-queryclient-instance